crates/experimentation_platform/src/api/experiments/handlers.rs

```diff
@@ -265,7 +273,12 @@ async fn create(
     let inserted_experiment: Experiment = inserted_experiments.remove(0);
     let response = ExperimentCreateResponse::from(inserted_experiment);
 
-    Ok(Json(response))
+    let mut http_resp = HttpResponse::Ok();
+    config_version_id.and_then(|v| {
+        http_resp.insert_header((AppHeader::XConfigVersion.to_string(), v.clone()));
+        Some(v)
+    });
+    Ok(http_resp.json(response))
```

Comment:
This block is repeatedly used everywhere. Can be made it into a function.

Reply:
it  is dynamic with respect to handler , so it wouldn't reduce repetition , also this gives flexibility to add multiple headers and tweak httpresponse , so its fine

Reply:
@pratikmishra356 You are inserting `AppHeader::XConfigVersion` in the handlers response. You can do that through a fucntion also by passing the `version_id` to it. Correct me if i am missing something.

Reply:
No you are right , we can move config_version insertion in a function, 
but `http_resp.json(response)` should be kept inside the handler only